### PR TITLE
Refactor/state machine args

### PR DESCRIPTION
**Pull request by agutenkunst**
## Description
This is a minor step towards more readability. StateMachineArgs was introduced due to missing the existence of `

During Review please check with real hardware!

### Things to add, update or check by the maintainers before this PR can be merged.

- [x] ~Public api function documentation~
- [x] Architecture documentation reflects actual code structure
- [x] ~[Tutorials](https://wiki.ros.org/pilz_robots/Tutorials/)~
- [x] ~Overview on [ROS wiki](https://wiki.ros.org/pilz_robots)~
- [x] ~Package Readme ([example pilz_robots](https://github.com/PilzDE/pilz_robots/blob/melodic-devel/README.md))~
- [x] Good commit messages ([some tips](https://dev.to/jacobherrington/how-to-write-useful-commit-messages-my-commit-message-template-20n9))
- [x] CHANGELOG.rst updated
- [x] Copyright headers
- [x] ~Examples~

### Review Checklist
- [x] ~Soft- and hardware architecture (diagrams and description)~
- [x] Test review (test plan and individual test cases)
- [x] ~Documentation describes purpose of file(s) and responsibilities~
- [x] Code (coding rules, style guide)

### Release planning (please answer)
- [x] ~When is the new feature released?~
- [x] ~Which dependent packages do have to be released simultaneously?~

### Hardware tests
- [x] Perform hardware tests

### CODE REVIEW

**Comment by rfeistenauer** [comment]
Manual hardware test was successfull

**Review by cdoehn — CHANGES_REQUESTED** [review]
suggestions:
* remove the abbreviations CB, cb and replace them by the full word callback and Callback.
* link to MSM docu (see comment)
* introduce a TimeoutHandler data type (see comment)

File: standalone/include/psen_scan_v2_standalone/protocol_layer/scanner_state_machine_def.h [hunk 1]
```diff
@@ -14,11 +14,38 @@
...
 #include "psen_scan_v2_standalone/data_conversion_layer/start_request_serialization.h"
+#include "psen_scan_v2_standalone/scanner_configuration.h"
+#include "psen_scan_v2_standalone/communication_layer/udp_client.h"
 namespace psen_scan_v2_standalone
 {
 namespace protocol_layer
 {
-inline ScannerProtocolDef::ScannerProtocolDef(StateMachineArgs* const args) : args_(args)
+inline ScannerProtocolDef::ScannerProtocolDef(const ScannerConfiguration config,
+                                              const communication_layer::NewDataHandler& control_data_handler,
```
[the remark this answers is not in the document]
Comment (agutenkunst) [reply standalone/include/psen_scan_v2_standalone/protocol_layer/scanner_state_machine_def.h:24]: I choose to rename almost all handlers that are actually callbacks (no class) into `callback`, hope you are fine with that.

File: standalone/include/psen_scan_v2_standalone/protocol_layer/scanner_state_machine_def.h [hunk 2]
```diff
@@ -14,11 +14,38 @@
...
+                                              const communication_layer::ErrorHandler& control_error_handler,
+                                              const communication_layer::NewDataHandler& data_data_handler,
```
[the remark this answers is not in the document]
Comment (agutenkunst) [reply standalone/include/psen_scan_v2_standalone/protocol_layer/scanner_state_machine_def.h:26]: I choose to rename almost all handlers that are actually callbacks (no class) into `callback`, hope you are fine with that.

**Review by cdoehn — APPROVED** [review]

**Comment by agutenkunst** [comment]
@cdoehn ready to merge?

**Pull request merged**